core/src/saros/communication/info/InfoManager.java

```diff
@@ -0,0 +1,179 @@
+package saros.communication.info;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.log4j.Logger;
+import org.jivesoftware.smack.PacketListener;
+import org.jivesoftware.smack.filter.AndFilter;
+import org.jivesoftware.smack.packet.IQ;
+import org.jivesoftware.smack.packet.Packet;
+import saros.annotations.Component;
+import saros.communication.extensions.InfoExchangeExtension;
+import saros.context.IContextKeyBindings.SarosVersion;
+import saros.net.IReceiver;
+import saros.net.ITransmitter;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.contact.IContactsUpdate;
+import saros.net.xmpp.contact.IContactsUpdate.UpdateType;
+import saros.net.xmpp.contact.XMPPContact;
+import saros.net.xmpp.contact.XMPPContactsService;
+
+/**
+ * Component for figuring out whether two Saros plug-in instances with known Version are compatible
+ * and to exchange Infos outside of Sessions.
+ *
+ * <p>This class compares if local and remote version (not checking qualifier) are the same.
+ */
+@Component(module = "core")
+public class InfoManager {
```

Comment:
What do you think about splitting the class into two classes:
* InfoManager providing the general methods for exchanging information
* VersionManager using the InfoManager to provide methods for exchanging the versions and determining compatibility

Reply:
I would suggest the same as.

Reply:
Yes, that's a good idea, and I've changed it accordingly.